Tests/OptimizelyTests-DataModel/OptimizelyJSONTests.swift

```diff
@@ -0,0 +1,489 @@
+        } catch {
+            err = error
+        }
+        XCTAssertNil(err)
+        XCTAssertTrue(NSDictionary(dictionary: optimizelyJSONToMap).isEqual(to: self.data))
+    }
+    
+    func testToString() {
+        var err: Error?
+        var optimizelyJSONToString: String!
+        do {
+            try optimizelyJSONToString = self.optimizelyJSON.toString()
+        } catch {
+            err = error
+        }
+        XCTAssertNil(err)
+        XCTAssertEqual(optimizelyJSONToString, self.payload)
+    }
+    
+    func testGetValueForInvalidJSONKeyAndEmptySchema() {
+        var err: Error?
+        var schema = EmptySchema()
+        do {
+            try self.optimizelyJSON.getValue(jsonPath: "field4.", schema: &schema)
+        } catch {
+            err = error
+        }
+        XCTAssertEqual(err?.localizedDescription, OptimizelyError.valueForKeyNotFound("").reason)
+    }
+    
+    func testGetValueForMissingJSONKeyAndEmptySchema() {
+        var err: Error?
+        var schema = EmptySchema()
+        do {
+            try self.optimizelyJSON.getValue(jsonPath: "some_key", schema: &schema)
+        } catch {
+            err = error
+        }
+        XCTAssertEqual(err?.localizedDescription, OptimizelyError.valueForKeyNotFound("some_key").reason)
+    }
+    
+    func testGetValueForInvalidJSONMultipleKeyAndEmptySchema() {
+        var err: Error?
+        var schema = EmptySchema()
+        do {
+            try self.optimizelyJSON.getValue(jsonPath: "field3.some_key", schema: &schema)
+        } catch {
+            err = error
+        }
+        XCTAssertEqual(err?.localizedDescription, OptimizelyError.valueForKeyNotFound("some_key").reason)
+    }
+    
+    func testGetValueForValidJSONKeyAndEmptySchema() {
+        var err: Error?
+        var schema = EmptySchema()
+        do {
+            try self.optimizelyJSON.getValue(jsonPath: "field4", schema: &schema)
+        } catch {
+            err = error
+        }
+        XCTAssertNil(err)
+    }
+    
+    func testGetValueForValidJSONKeyAndIncorrectSchema() {
+        var err: Error?
+        var schema = IncorrectSchema()
+        do {
+            try self.optimizelyJSON.getValue(jsonPath: "field4", schema: &schema)
+        } catch {
+            err = error
+        }
+        XCTAssertEqual(err?.localizedDescription, OptimizelyError.failedToAssignValueToSchema.reason)
+        
+    }
+    
+    func testGetValueForValidJSONMultipleKeyAndEmptySchema() {
+        var err: Error?
+        var schema = EmptySchema()
+        do {
+            try self.optimizelyJSON.getValue(jsonPath: "field4.inner_field1", schema: &schema)
+        } catch {
+            err = error
+        }
+        XCTAssertEqual(err?.localizedDescription, OptimizelyError.failedToAssignValueToSchema.reason)
+    }
+    
+    func testGetValueForValidJSONMultipleKeyAndNonDecodableSchema() {
+        var err: Error?
+        var schema: [Any] = []
```

Comment:
Don't provide invalid schema, there msut be a valid schema and it should be non-decodable. may be a simple schema.